staging/src/k8s.io/apiserver/pkg/storage/etcd3/preflight/checks.go

```diff
@@ -36,7 +36,11 @@ func (EtcdConnection) serverReachable(connURL *url.URL) bool {
 	if scheme == "http" || scheme == "https" || scheme == "tcp" {
 		scheme = "tcp"
 	}
-	if conn, err := net.DialTimeout(scheme, connURL.Host, connectionTimeout); err == nil {
+	addr := connURL.Host
```

Comment:
What's the purpose of this change. 

etcd does not support the path component when its opening the listening socket:

```
% ./bin/etcd --listen-client-urls=unix://192.168.17.3:2000/foo.etcd --advertise-client-urls=unix://192.168.17.3:2000/foo.etcd
invalid value "unix://192.168.17.3:2000/foo.etcd" for flag -listen-client-urls: URL must not contain a path: unix://192.168.17.3:2000/foo.etcd
```
This is validated here: 
```
https://github.com/etcd-io/etcd/blob/a3174d0f8ec6ec58827d7d86448bb4df08ae69e4/pkg/types/urls.go#L46
```

@gyuho I think etcd tests would highly benefit from ability to connect to 
```unix:///tmp/etcdTestSuite.testSuccessfulConnection.socket```
so utilize naming schema of grpc: 
  https://github.com/grpc/grpc/blob/master/doc/naming.md

I don't know the historical reason, but it's not obvious choice to change it in 3.5 and 
to what degree make it backward compatible.
  - etcd <=3.4 perspective:  unix://192.168.0.15:2345 is relative path (./192.168.0.15:2345)
  - grpc perspective unix://192.168.0.15:2345 ->  '192.168.0.15:2345' is considered a global path.
    `unix:192.168.0.15:2345` would be relative path.

Please open issue in etcd component with description of problem you are solving.

Reply:
Sorry, I should have mentioned that there is a necessary etcd fix, https://github.com/etcd-io/etcd/pull/12469, which solved the validation you mentioned.

> Please open issue in etcd component with description of problem you are solving.
 
I opened https://github.com/kubernetes/kubernetes/issues/97325.

In short, with https://github.com/etcd-io/etcd/pull/12469 and this PR, the kube-apiserver can talk to the etcd on the same machine via UDS with flag `--etcd-servers=unix:///tmp/etcd.socket`.